external-api/src/v1/controllers/companies.controller.ts

```diff
@@ -0,0 +1,46 @@
+import axios, { AxiosError, HttpStatusCode } from 'axios';
+import { Request, Response } from 'express';
+import * as dotenv from 'dotenv';
+import { isValidCompanyRegistrationNumber } from '@ukef/dtfs2-common';
+
+dotenv.config();
+
+const { APIM_MDM_KEY, APIM_MDM_VALUE, APIM_MDM_URL } = process.env;
+const headers = {
+  'Content-Type': 'application/json',
+  [String(APIM_MDM_KEY)]: APIM_MDM_VALUE,
+};
+
+/**
+ * Get company details for the company matching the Companies House registration number provided.
+ * @param {Request} req Express request with `registrationNumber` as a URL parameter.
+ * @param {Response} res Express response.
+ * @returns {Promise<Response>} Express response with `status` and `data`. `Data` contains the company details,
+ * or error data if the request was unsuccessful.
+ */
+export const getCompanyByRegistrationNumber = async (req: Request, res: Response): Promise<Response> => {
+  let response: { status: number | undefined; data: unknown };
+
+  const { registrationNumber } = req.params;
+
+  if (!isValidCompanyRegistrationNumber(registrationNumber)) {
+    response = {
+      status: 400,
```

Comment:
Http Status codes please for all the commit files.

Reply:
Done.